File: app/src/main/java/com/example/medplus/model/Appointment.kt
package com.example.medplus.model

import com.google.firebase.Timestamp
import com.google.gson.annotations.SerializedName

/**
 * Data model for a patient appointment in MedPlus.
 */
data class Appointment(
    @SerializedName("_id") val appointmentId: String = "",
    val patientId: String = "",
    val patientName: String = "",
    val doctorId: String = "",
    val doctorName: String = "",
    val department: String = "",
    val date: String = "",
    val time: String = "",
    val status: String = "UPCOMING",
    val tokenNumber: String? = null,
    val createdAt: Timestamp = Timestamp.now(),
    val timestamp: Timestamp? = null, // For sorting and filtering
    val consultationStartedAt: String? = null,
    val consultationCompletedAt: String? = null
)

File: backend/models/Appointment.js
const mongoose = require('mongoose');

const AppointmentSchema = new mongoose.Schema({
  patientId: { type: String, required: true },
  patientName: { type: String, required: true },
  doctorId: { type: String, required: true },
  doctorName: { type: String, required: true },
  department: { type: String, default: "" },
  date: { type: String, required: true }, // Format: YYYY-MM-DD
  time: { type: String, required: true }, // Format: HH:MM
  status: { type: String, enum: ['UPCOMING', 'IN_PROGRESS', 'COMPLETED', 'CANCELLED'], default: 'UPCOMING' },
  tokenNumber: { type: String, default: null },
  createdAt: { type: Date, default: Date.now },
  timestamp: { type: Date }, // Date object constructed from date & time
  consultationStartedAt: { type: Date },
  consultationCompletedAt: { type: Date }
});

module.exports = mongoose.model('Appointment', AppointmentSchema);
